tests/compile-errors/src/lib.rs

```diff
@@ -10,5 +10,7 @@ mod tests {
         t.compile_fail("src/unsupported-type.rs");
         t.compile_fail("src/no-closure.rs");
         t.compile_fail("src/no-provider-file.rs");
+        t.compile_fail("src/zero-arg-probe-type-check.rs");
+        t.compile_fail("src/different-serializable-type.rs");
```

Comment:
What do you think of having a directory of these tests as in dropshot?

https://github.com/oxidecomputer/dropshot/blob/main/dropshot/tests/fail.rs

Reply:
I think that's fine, but I'm going to defer this one.